src/main/java/com/google/sps/data/ScheduledCourse.java

```diff
@@ -38,60 +35,26 @@
    */
   public ScheduledCourse(String name, String courseID, String subject, float credits, 
       boolean isRequired, Section section) {
-    this.name = name;
-    this.courseID = courseID;
-    this.subject = subject;
-    this.credits = credits;
-    this.isRequired = isRequired;
+    super(name, courseID, subject, credits, isRequired, Arrays.asList(section));
     this.section = section;
   }
 
-  public String getName() {
-    return this.name;
-  }
-
-  public String getCourseID() {
-    return this.courseID;
-  }
-
-  public String getSubject() {
-    return this.subject;
-  }
-
-  public float getCredits() {
-    return this.credits;
-  }
-
-  public boolean isRequired() {
-    return this.isRequired;
-  }
-
-  public Section getSection() {
-    return this.section;
-  }
-
   @Override
   public boolean equals(Object other) {
-    if (other == this) {
-      return true;
-    }
-
-    if (!(other instanceof ScheduledCourse)) {
-      return false;
+    if (other instanceof ScheduledCourse) {
+      return super.equals(other) && this.section.equals(((ScheduledCourse) other).getSection());
     }
 
-    ScheduledCourse otherCourse = (ScheduledCourse) other;
-    return otherCourse.name.equals(this.name) 
-        && otherCourse.courseID.equals(this.courseID)
-        && otherCourse.subject.equals(this.subject) 
-        && Math.abs(otherCourse.credits - this.credits) < 0.01
-        && otherCourse.isRequired == this.isRequired 
-        && otherCourse.section.equals(this.section);
+    return false;
   }
 
   @Override
   public int hashCode() {
-    return Objects.hash(name, courseID, subject, credits, isRequired, section);
+    return super.hashCode();
+  }
```

Comment:
You're right, you don't need to override `hashCode()` if you're just calling `super.hashCode()` since it will inherit that method. This can be removed.

I'm curious though since you adjusted the `equals()` method to only allow equality between two ScheduledCourses and not a ScheduledCourse and a Course, is it ok for a ScheduledCourse and a Course to hash to the same value?

Reply:
You're right! Ideally they wont. Addressed this issue.